internal/docs/mdx.go

```diff
@@ -0,0 +1,49 @@
+package docs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/spf13/cobra"
+)
+
+func GenMdxFile(cmd *cobra.Command, dir string) error {
+	for _, c := range cmd.Commands() {
```

Comment:
Could we build the tree of the commands at the same place between the two version? Many of those lines are duplicated.

Maybe in the `docs.go`, adding a `getCommands` like
```go
func getCommands(cmd *cobra.Command) []Command {
	var commands []Command
	for _, c := range cmd.Commands() {
		if c.Hidden || c.Name() == "help" {
			continue
		}
		command := newCommand(c)
		if c.HasAvailableSubCommands() {
			for _, sub := range c.Commands() {
				command.SubCommands = append(command.SubCommands, newCommand(sub))
			}
		}
		commands = append(commands, command)
	}

	return commands
}
```